src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -599,8 +632,17 @@ def _visit_map(self, node: gtir.FunCall) -> DataExpr:
         fun_python_code = gtir_python_codegen.get_source(fun_node)
         tasklet_expression = f"{output_connector} = {fun_python_code}"
 
-        # TODO(edopao): extract offset_dim from the input arguments
-        offset_dim = gtx_common.Dimension("", gtx_common.DimensionKind.LOCAL)
+        input_local_offsets = [
+            input_arg.local_offset for input_arg in input_args if input_arg.local_offset is not None
+        ]
+        if len(input_local_offsets) == 0:
+            raise ValueError(f"Missing information on local dimension for map node {node}.")
+
+        local_offset = input_local_offsets[0]
+        offset_provider = self.subgraph_builder.get_offset_provider(local_offset)
+        assert isinstance(offset_provider, gtx_common.Connectivity)
+        local_size = offset_provider.max_neighbors
```

Comment:
This might be a very naïve question, but assume you have a grid were every point has 6 neighbours, then 10 is a valid value of `_max_neighbors`, although it is not a very good one.
Before you were inspecting the results, so it might be possible that you found that you only need 6.
So I would have have kept the old inspection, and if I did not found a result at the end used the max value.

However, I am not sure how realistic this situation is and if I explained it well enough (I am bad at that).